aws/sdk/sync-models.py

```diff
@@ -21,44 +21,71 @@ def discover_aws_models():
     else:
         return None
 
-# Acquire model location
-aws_models_repo = discover_aws_models()
-if aws_models_repo == None:
-    if len(sys.argv) != 2:
-        print("Please provide the location of the aws-models repository as the first argument")
-        sys.exit(1)
-    else:
-        aws_models_repo = sys.argv[1]
+def discover_new_models(aws_models_repo, known_models):
+    new_models = []
+    for model in os.listdir(aws_models_repo):
+        if model not in known_models and path.exists(Path(aws_models_repo) / model / "smithy" / "model.json"):
+            new_models.append(model)
+    return new_models
 
-print("Copying over known models...")
-known_models = set()
-for model in os.listdir("aws-models"):
-    if not model.endswith('.json'):
-        continue
-    model_name = model[:-len('.json')]
-    known_models.add(model_name)
-    source_path = Path(aws_models_repo) / model_name / "smithy" / "model.json"
-    if not source_path.exists():
-        print(f"  Warning: cannot find model for '{model_name}' in aws-models, but it exists in smithy-rs")
-        continue
-    # Add a newline at the end when copying the model over
+# HACK: Modify the S3 model directly to change the type of `Size` until Smithy
+# allows this transformation directly (https://github.com/awslabs/smithy/pull/900).
+def hack_s3_model(model_path):
+    result = subprocess.run(["jq", "-M", ".shapes[\"com.amazonaws.s3#Size\"].type = \"long\"", str(model_path)], capture_output = True, check = True)
```

Comment:
if we do this in the future, might be clearer to just do it in Python and mutate the JSON